Marlin/src/lcd/ultralcd.cpp

```diff
@@ -764,54 +759,55 @@ void MarlinUI::update() {
     update_buttons();
 
     // If the action button is pressed...
-    static bool wait_for_unclick; // = 0
-    auto generate_click = [&]() {
-      if (!wait_for_unclick) {                        // If not waiting for a debounce release:
-        wait_for_unclick = true;                      //  - Set debounce flag to ignore continous clicks
-        lcd_clicked = !wait_for_user && !no_reentry;  //  - Keep the click if not waiting for a user-click
-        wait_for_user = false;                        //  - Any click clears wait for user
-        quick_feedback();                             //  - Always make a click sound
-      }
-    };
+    static bool wait_for_unclick; // = false
 
     #if ENABLED(TOUCH_BUTTONS)
-      if (touch_buttons) {
-        if (buttons & EN_C)
-          generate_click();
-        else if (buttons & (EN_A | EN_B)) {             // Ignore the encoder if clicked, to prevent "slippage"
-          const millis_t ms = millis();
-          if (ELAPSED(ms, next_button_update_ms)) {
-            next_button_update_ms = ms + 50;
-            encoderDiff = (ENCODER_STEPS_PER_MENU_ITEM) * (ENCODER_PULSES_PER_STEP);
-            if (buttons & EN_A) encoderDiff *= -1;
-            if (!wait_for_unclick) {
-              next_button_update_ms += 250;
+
+      static bool arrow_pressed; // = false
+
+      // Handle touch_buttons which are slow to read
+      if (ELAPSED(ms, next_button_update_ms)) {
+        const uint8_t touch_buttons = touch.read_buttons(),
+                      touch_arrows = touch_buttons & (EN_A | EN_B);
+        if (touch_buttons) {
+          if (touch_buttons & EN_C) buttons |= EN_C;
+          #if HAS_LCD_MENU && LCD_TIMEOUT_TO_STATUS
+            return_to_status_ms = ms + LCD_TIMEOUT_TO_STATUS;
+          #endif
+          if (touch_arrows) {
+            next_button_update_ms = ms + 50;       // Set delay for repeat
+            encoderDiff = (ENCODER_STEPS_PER_MENU_ITEM) * (ENCODER_PULSES_PER_STEP) * encoderDirection;
```

Comment:
This value should be different when navigating a menu versus when editing a value. But perhaps on the touch display the "pulses per step" is always set to 1…?

Reply:
Hmm… By default the `ENCODER_PULSES_PER_STEP` falls back to 5. If the `TOUCH_BUTTONS` are the only encoder, then this can be set to 1 to slightly simplify the code. Is it possible for `TOUCH_BUTTONS` to exist alongside a regular click-wheel encoder?

Reply:
@thinkyhead , I propose this code. The reason is to prepare full area touchscreen

Reply:
A full touch-screen is going to need an entirely different approach and won't go through this "button" nonsense. Sorry about trouncing `BTN_D`. An oversight.

Reply:
At present the vertical touch coordinate alone isn't going to very useful since we have a variety of screen layouts in Marlin. Every single screen layout type will need to have its own method to interpret touches and pass them on to the correct handlers. In cases where we only have a touch screen and no other buttons, we don't need to have up/down buttons on the screen unless something will actually be moved up/down. For menus, it will be better to have square buttons instead of textual menu items where possible, and textual menu items will need to be spaced apart more so that touches are less likely to invoke the wrong item. It will be better to write a new handler for these events rather than keep hacking everything on top of the rotary encoder.

Reply:
maybe menus could be scrolled, and -/+ slided for values, but just ideas ;)

Reply:
Would you prefer that "⬆️➕" and "⬇️➖" always stick together?

Reply:
the up/down are for menus... if menu can be slided, the arrows are no more useful, i mean...

Reply:
I would like it if the arrows appeared when the buttons act like arrows (pretty much only in menus). And +/- should appear when the buttons act to change values (value editing screens). This might make the graphics a bit cleaner.

Reply:
(In English we say "scroll" rather than "slide" for views that shift around within a box.)

Reply:
sorry, ive the "slider" picture in my head

Reply:
https://youtu.be/ClhTLofXOnA Check this out.

Reply:
very good yep, but as i said before, this is not the upscale dogm anymore...

Reply:
I like the bare-bones text in a simple line-box for the menu items. Color-coding by type is also cool, where yellow might mean "editable," and so on. If there were more buttons than could fit on the screen, it would get a pager, I suppose. But the nice thing about such a bare-bones approach is that it may encourage people to want to make it more interesting. But can it do all the languages?

Reply:
> https://youtu.be/ClhTLofXOnA Check this out.

Hi Can you provide the config files for MKS robin TFT